utils/s3/client.cc

```diff
@@ -89,6 +90,19 @@ future<> ignore_reply(const http::reply& rep, input_stream<char>&& in_) {
 client::client(std::string host, endpoint_config_ptr cfg, semaphore& mem, global_factory gf, private_tag, std::unique_ptr<aws::retry_strategy> rs)
         : _host(std::move(host))
         , _cfg(std::move(cfg))
+        , _creds_invalidation_timer([this] {
+            s3l.info("Invalidating creds");
+            _credentials = {};
+        })
+        , _creds_update_timer([this] {
```

Comment:
BTW, when the client dies with this timer's lambda running the `_credentials` assignment may happen to become use-after-free

Reply:
[Cancelling timer](https://github.com/scylladb/scylladb/pull/21830/files#diff-28c1d03a5facd99f25f466e6bae94c0fab2bb8845c0538dd489db7832505e884R1358) in the `close` is enough?

Reply:
Unfortunately no. timer.cancel() only makes sure the time lambda itself is not running, while this timer's lambda spawns another one, which is out of this timer's reach